panther_description/launch/rviz.launch.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+
+# Copyright 2020 ros2_control Development Team
+# Copyright 2024 Husarion sp. z o.o.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from launch import LaunchDescription
+from launch.actions import DeclareLaunchArgument
+from launch.conditions import IfCondition
+from launch.substitutions import (
+    EnvironmentVariable,
+    LaunchConfiguration,
+    PathJoinSubstitution,
+    PythonExpression,
+)
+from launch_ros.actions import Node, SetParameter
+from launch_ros.substitutions import FindPackageShare
+from nav2_common.launch import ReplaceString
+
+
+def generate_launch_description():
+
+    namespace = LaunchConfiguration("namespace")
+    declare_namespace_arg = DeclareLaunchArgument(
+        "namespace",
+        default_value=EnvironmentVariable("ROBOT_NAMESPACE", default_value=""),
+        description="Add namespace to all launched nodes.",
+    )
+
+    rviz_config = LaunchConfiguration("rviz_config")
+    declare_rviz_config_arg = DeclareLaunchArgument(
+        "rviz_config",
+        default_value=PathJoinSubstitution(
+            [FindPackageShare("panther_description"), "rviz", "husarion_ugv.rviz"]
+        ),
+        description="RViz configuration file.",
+    )
+
+    use_rviz = LaunchConfiguration("use_rviz")
+    declare_use_rviz_arg = DeclareLaunchArgument(
+        "use_rviz",
+        default_value="True",
+        description="Run RViz simultaneously.",
+        choices=["True", "true", "False", "false"],
+    )
```

Comment:
This may seem strange, but we already have it done in other packages as well (on latest ros2-lynx-devel). Since the arguments are passed to the parent launch file, there is no need to specify this parameter in bringup.launch.py ​​and simulation.launch.py, which improves the readability of these two most important files.

The launch files are not ideal, but imo this solution is cleaner and reduce redundancy of declarated argument.

Reply:
That's not my point, this argument should be on simulation launch file level. There you decide if you should launch rviz or not, not inside rviz launching file. Why would you ever want to use rviz.launch file with argument that says "don't launch it"?
Picture that, If you go to simulation file you will see that rviz is called without any arguments so you assume that it is going to be launched. You don't have any indication that it should be otherwise. It's good to have arguments that are rviz specific "hidden" in simulation.launch, but this one refers to the way the simulation should be launched.

Reply:
Ok, I made sure about that